v2/coord/coordinator.go

```diff
@@ -324,7 +324,7 @@ func (c *Coordinator) GetClosestNodes(ctx context.Context, k KadKey, n int) ([]N
 	closest := c.rt.NearestNodes(k, n)
 	nodes := make([]Node, 0, len(closest))
 	for _, id := range closest {
-		nh, err := c.networkBehaviour.getNodeHandler(ctx, NodeIDToPeerID(id))
+		nh, err := c.networkBehaviour.getNodeHandler(ctx, peer.ID(id))
```

Comment:
from this call, it looks liek `getNodeHandler` is operating on a `peer.ID`.

As a general separation, I think it would be great if most of the things in `coord` only operated on `kadt.PeerID`. Not sure if there is anything prohibiting that in this case here - just a general idea.

Reply:
Yes, that sounds sensible, will check usage here before merging

Reply:
I have removed this specific usage and converted the self fields for the coordiinator and behaviours to kadt.PeerID, but doing more means touching the Router interface so let's do that in a later PR.